plugins/kubernetes/app/models/kubernetes/namespace.rb

```diff
@@ -11,5 +11,14 @@ class Namespace < ActiveRecord::Base
     has_many :projects, dependent: nil, foreign_key: :kubernetes_namespace_id, inverse_of: :kubernetes_namespace
 
     validates :name, presence: true, uniqueness: true, format: NAME_PATTERN
+    before_destroy :ensure_unused
+
+    private
+
+    def ensure_unused
+      return if projects.none?
+      errors.add :base, 'Can only delete when not used by any project.'
```

Comment:
the verbage is a little strange here.. would have expected `"Used by #{projects.size} projects."`